raiden_contracts/deploy/etherscan_verify.py

```diff
@@ -0,0 +1,191 @@
+import json
+import click
+import requests
+from time import sleep
+from pathlib import Path
+
+from eth_abi import encode_abi
+
+from raiden_contracts.constants import (
+    CONTRACT_ENDPOINT_REGISTRY,
+    CONTRACT_SECRET_REGISTRY,
+    CONTRACT_TOKEN_NETWORK_REGISTRY,
+)
+from raiden_contracts.contract_manager import (
+    ContractManager,
+    contracts_precompiled_path,
+    contracts_source_path,
+    get_contracts_deployed,
+)
+
+
+@click.command()
+@click.option(
+    '--chain-id',
+    default=3,
+    help='Chain id. E.g. --chain-id 3',
+)
+@click.option(
+    '--apikey',
+    required=True,
+    help='A valid Etherscan APIKEY is required.',
+)
+@click.option(
+    '--guid',
+    help='GUID from a previous verification attempt. Tries to get the submission status.',
+)
+@click.option(
+    '--contract-name',
+    default=None,
+    help='Contract name. Options: EndpointRegistry | SecretRegistry | TokenNetworkRegistry',
+)
+def etherscan_verify(chain_id, apikey, guid, contract_name):
+    contract_manager = ContractManager(contracts_precompiled_path())
+    source_path = contracts_source_path()
+
+    deployment_info = get_contracts_deployed(chain_id)
+
+    if chain_id == 3:
+        etherscan_api = 'https://api-ropsten.etherscan.io/api'
+    elif chain_id == 4:
+        etherscan_api = 'https://api-rinkeby.etherscan.io/api'
+    elif chain_id == 42:
+        etherscan_api = 'https://api-kovan.etherscan.io/api'
+    elif chain_id == 1:
+        etherscan_api = 'https://api.etherscan.io/api'
+
+    if guid:
+        guid_status(etherscan_api, guid)
+        return
+
+    if contract_name is None or contract_name == CONTRACT_ENDPOINT_REGISTRY:
+        path = source_path['raiden'].joinpath(f'{CONTRACT_ENDPOINT_REGISTRY}.sol')
+        source = path.read_text()
+        constructor_args = ''
+        etherscan_verify_contract(
+            etherscan_api,
+            apikey,
+            contract_manager.contracts[CONTRACT_ENDPOINT_REGISTRY],
+            CONTRACT_ENDPOINT_REGISTRY,
+            deployment_info['contracts'][CONTRACT_ENDPOINT_REGISTRY],
+            source,
+            constructor_args,
+        )
+
+    if contract_name is None or contract_name == CONTRACT_SECRET_REGISTRY:
+        path = source_path['raiden'].joinpath(f'{CONTRACT_SECRET_REGISTRY}.sol')
+        source = path.read_text()
+        constructor_args = ''
```

Comment:
Can this be refactored somehow into it's own function? as this block uses the same code as the block above it.